### ISSUE TITLE
Move Mollie back to Payments API instead of Orders API

### ISSUE THREAD

**Issue by martijnvdbrug** [vendure-ecommerce/vendure#2921]
**Is your feature request related to a problem? Please describe.**

> A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
I am always frustrated when payment providers tell me to do one thing and a year later want me to move back to the original solution...

Ok, now, serious: The Mollie docs mention that [they do not recommend moving to the orders API anymore](https://docs.mollie.com/reference/orders-api). Something that we fixed 1,5 year ago in [this issue](https://github.com/vendure-ecommerce/vendure/issues/1824).

The main issue with the payments API was that it does not support some payment methods like Klarna. The docs state that this will be implemented somewhere this year. **We should not move to the payments API before this is implemented**

From the [Mollie Discord](https://discord.com/channels/1037712581407817839/1037752322585018368/1243549435514851380):

> How does it differ from the Orders API?
> It is good to know that on the Payments API, the lines are simply meta-data. In contrast to the Orders API, line items do not have any business logic related to it (like a status). You also cannot create a refund via the API only referencing a line item, you simply send in an amount. This is on purpose since we realised from the Orders API that building logic on line items is actually making things more complex and less flexible.

**Describe the solution you'd like**

When Mollie implements the missing payment methods in the Payments API, there really is no reason to use the Orders API anymore, which simplifies the code on our side quite a bit.

**Pro's**
* We don't have to update orders in Mollie when a checkout is called twice. We just create a new payment.
* Arbitrary refund amounts don't have to be linked to order lines anymore. We can just refund an amount from a given payment
* [No more rounding workarounds](https://github.com/vendure-ecommerce/vendure/issues/1939#issuecomment-1362962133). Mollie uses order line rounding where Vendure uses unit rounding
* Architecture: It makes sense to have Vendure responsible for its order lines and corresponding amounts and taxes, and Mollie just for any arbitrary amount to be paid, calculated by Vendure. With the Orders API we have to replicate a Vendure order in Mollie, and calculations (like taxes) have to match.

**Con's**
I can't see any, but I would like to ask any Mollie users: What do you use from the Mollie Order's API that is not supported by the Payments API?

**Comment by casperiv0** [comment]
I'm not seeing any drawbacks/cons to this either. Makes the entire logic much easier to understand too. I've asked our team if we use any of those services that are coming later this year in Mollie (Updating this comment once I have that information.)

**Comment by martijnvdbrug** [comment]
@casperiv0 Thanks for the feedback! We have several clients who are using the pay-later (e.g. Klarna and others) methods, so I propose if we move back to the payments API, we only do so after the Payments API supports those methods as well.

**Comment by dlhck** [comment]
@martijnvdbrug do you have a timeline on that?

**Comment by martijnvdbrug** [comment]
@dlhck Not really, we are dependent on Mollie: they are currently migrating all Payment methods to the Payments API. Currently some methods like Klarna are still only available with the Orders API, so we cant migrate the Vendure plugin yet. Their estimate was 'later this year', but no news yet.

**Comment by martijnvdbrug** [comment]
@dlhck The mollie "orders API" is officially deprecated. I am now working on a PR to make the plugin use the payments API again: https://docs.mollie.com/docs/migrating-from-orders-to-payments

**Comment by dlhck** [comment]
@martijnvdbrug thank you!

**Comment by martijnvdbrug** [comment]
Closed by #3603
